### PR TITLE
feat: generate data for spread operator

### PR DESCRIPTION
**Pull request by EarthyOrange**

### CODE REVIEW

**Comment by EarthyOrange** [comment]
@gr0uch I ran in to issues with doc-tree still unable to parse the spread operator. I have added code to massage the 'SpreadElement' Node data to look like what the foreach block expects. I have also added the spreadProperty boolean property to the context object, just like the generator function.

Please go through the PR and let me know your thoughts.

**Comment by EarthyOrange** [comment]
@gr0uch Can you please take a look?

File: lib/walk_nodes.js [hunk 1]
```diff
@@ -148,14 +148,35 @@ module.exports = function walkNodes (AST) {
...
-              type: fnTypes.has(property.value.type) ?
-                names.method : names.property
-            }, declaration.id ? {
-              target: declaration.id.name
-            } : null)))
+          declaration.init.properties
+            .map(property => {
+              if (property.type === 'SpreadElement')
+                return Object.assign(
+                  property,
```
Comment (gr0uch) [inline lib/walk_nodes.js:155]: careful, this is mutating the original `property`, are you sure?

**Review by gr0uch — COMMENTED** [review]
The tests look good. I just have one concern whether it's safe to mutate `property`. Might want to do `Object.assign({}, ...)` here.

File: lib/walk_nodes.js (hunk 1 above)
Comment (EarthyOrange) [reply lib/walk_nodes.js:155]: Wow! Thanks for catching this!

**Review by gr0uch — APPROVED** [review]

**Comment by gr0uch** [comment]
thanks, i will publish again :)

**Pull request merged**